### ISSUE TITLE
Expose platform webview for initialization to enable customization

### ISSUE THREAD

**Issue by Eilon** [Eilon/MauiHybridWebView#45]
For example, in https://github.com/Eilon/MauiHybridWebView/issues/27#issuecomment-1737182239 this was mentioned:

> Can I disable Android WebView TextZoom on this？ I can set it by use blazorWebView in the Init event:
> 
> ```
>                 private void blazorWebView_BlazorWebViewInitialized(object sender, Microsoft.AspNetCore.Components.WebView.BlazorWebViewInitializedEventArgs e)
>                 {
>         #if ANDROID
>                     e.WebView.Settings.TextZoom = 100;
>                    e.WebView.Settings.SetSupportZoom(false);
>         #endif
>                 }
> ```
> 
> but It seems private in your library =>private Android.Webkit.WebView PlatformWebView

**Comment by pagirre** [comment]
That would be great as I have had to modify a local copy of your codebase to add this tweaks in bold:

private partial Task InitializeHybridWebView()
{
    // Note that this is a per-app setting and not per-control, so if you enable
    // this, it is enabled for all Android WebViews in the app.
    AWebView.SetWebContentsDebuggingEnabled(enabled: EnableWebDevTools);            

    PlatformWebView.Settings.JavaScriptEnabled = true;
    **PlatformWebView.Settings.AllowUniversalAccessFromFileURLs = true;
    PlatformWebView.Settings.AllowFileAccessFromFileURLs = true;
    PlatformWebView.Settings.AllowFileAccess = true;
    PlatformWebView.Settings.AllowContentAccess = true;**

    _javaScriptInterface = new HybridWebViewJavaScriptInterface(this);
    PlatformWebView.AddJavascriptInterface(_javaScriptInterface, "hybridWebViewHost");

    return Task.CompletedTask;
}

Thank you for your effort, Eilon

**Comment by pagirre** [comment]
Hi again. 

I was convinced that doing these little tweaks(exposing those 5 fields just to check their value when navigating) in your code, 

![image](https://github.com/Eilon/MauiHybridWebView/assets/114993270/83b57efd-d021-44ce-a636-62daca6f6c78)

I would be able to avoid CORS (in the next screenshot you can check that all the abovementioned fields are set to true while Navigating event)

![image](https://github.com/Eilon/MauiHybridWebView/assets/114993270/1b007701-048e-470a-a309-bedbbb64653c)

But I am afraid I have not succeeded:

<img width="921" alt="image" src="https://github.com/Eilon/MauiHybridWebView/assets/114993270/67d0c94f-c07d-4a72-939c-3f11bbdcd76a">

Any help appreciated. Cheers!

**Comment by pagirre** [comment]
Forget about my last post. It is working now. 

The problem was that a colleague did not provide me the full information: the origin URI should have been localhost:3000 instead of localhost. That is what, I guess, is allowed to avoid CORS error.

But as you can see, AppHostAddress could be another member that I would like to set from outside at the beginning.

**Comment by Eilon** [comment]
This feature is merged for 1.0.0-preview6.

# Customizing web view initialization

To customize the platform web view you can add an event handler for the new `HybridWebViewInitialized` event:

```c#
    myHybridWebView.HybridWebViewInitialized += MyHybridWebView_WebViewInitialized;

    ...

    private void MyHybridWebView_WebViewInitialized(object sender, HybridWebViewInitializedEventArgs e)
    {
        // Use e.WebView to customize things
    }

```

And in that event handler you can access the platform-specific properties and methods of the web view control. This code would typically be in an `#if SOME_PLATFORM ... #endif` block because the APIs are different for each platform, and this enables platform-specific code to be compiled for only that platform.

For example, this event handler customizes the zoom settings on Windows and Android, and does nothing for iOS and macOS:

```c#
    private void MyHybridWebView_WebViewInitialized(object sender, HybridWebViewInitializedEventArgs e)
    {
#if WINDOWS
        // Disable the user manually zooming
        e.WebView.CoreWebView2.Settings.IsZoomControlEnabled = false;
#elif ANDROID
        // Customize zoom options
        e.WebView.Settings.TextZoom = 200;
        e.WebView.Settings.SetSupportZoom(false);
#endif
    }
```
